### DISCUSSION TITLE
how to support stroke and strokeWidth props in group?

### DISCUSSION THREAD

**Discussion by wuxingping** [fabricjs/fabric.js#8630] [General]
i want to set stroke and strokeWidth for group,what do i do?
egg:
var group = new fabric.group([],{
    stroke:"red",
    strokeWidth:10
})
it donot work  in current verison #master

**Comment by ShaMan123** [comment]
> it donot work in current verison #master

I that case your best option would be to add a rect to the group and set the stroke values on it

**Reply by ShaMan123** [reply]
In general that might be the best approach for v6 as well

**Reply by wuxingping** [reply]
it will be confused like that. for example when i drag the rect inside group,the another one inside group will not move.i mean the rect inside group will be not generate as a whole with group.i just hope group has stroke.

**Reply by wuxingping** [reply]
i found one version of v6 that a group has stroke.but no interactive for object inside group.it is not compatible with both? i mean the group has stroke and also the object inside group has interactive.

**Reply by wuxingping** [reply]
may i send you email to discuss more details? @ShaMan123

**Reply by ShaMan123** [reply]
my email is public in my profile

**Reply by wuxingping** [reply]
i send you this moring(ten hours ago),but no reply.my email is wxp67655049@126.com.   Looking forward to hearing from you

**Reply by ShaMan123** [reply]
I didn't receive an email.
Thinking of it the best approach would be overriding the drawObject method of Group.

**Reply by wuxingping** [reply]
[shacharnen@gmail.com](http://shacharnen@gmail.com/) this is your email?

**Reply by wuxingping** [reply]
i do send you email,maybe check your spam

**Reply by ShaMan123** [reply]
that is my email.
I didn't receive.
If it isn't proprietary data you are welcome to share here.

But this is the solution
> Thinking of it the best approach would be overriding the drawObject method of Group.

**Reply by ShaMan123** [reply]
https://github.com/fabricjs/fabric.js/blob/84b7a357cef0c899ecd6b3fc0a74de2d57283d21/src/shapes/Group.ts#L483-L489

**Reply by wuxingping** [reply]
ok, thank you so much,can you help me about discussion #8633.in fact.i am developing a very complex project by fabric. so i have many questions to ask. i am in China. network is not stable here.can you send me email? [wxp67655049@126.com](mailto:wxp67655049@126.com)there is my email address. i am looking forword more communication with you.
